crates/client/src/client/client_connection.rs

```diff
@@ -53,7 +63,11 @@ impl MessageFinalizer for Signer {
         time: u32,
     ) -> ProtoResult<(Vec<Record>, Option<MessageVerifier>)> {
         match self {
```

Comment:
I think you will encounter https://github.com/rust-lang/rust/issues/78123 here. Adding a `*` should be enough to fix it

Reply:
good call.